server/channels/store/sqlstore/channel_store.go

```diff
@@ -1489,24 +1489,38 @@ func (s SqlChannelStore) GetDeletedByName(teamId string, name string) (*model.Ch
 	return &channel, nil
 }
 
-func (s SqlChannelStore) GetDeleted(teamId string, offset int, limit int, userId string) (model.ChannelList, error) {
+func (s SqlChannelStore) GetDeleted(teamId string, offset int, limit int, userId string, skipTeamMembershipCheck bool) (model.ChannelList, error) {
 	channels := model.ChannelList{}
 
-	query := `
-		SELECT * FROM Channels
-		WHERE (TeamId = ? OR TeamId = '')
-		AND DeleteAt != 0
-		AND Type != ?
-		UNION
+	var query string
+	var args []interface{}
+
+	if skipTeamMembershipCheck {
+		query = `
 			SELECT * FROM Channels
 			WHERE (TeamId = ? OR TeamId = '')
 			AND DeleteAt != 0
-			AND Type = ?
-			AND Id IN (SELECT ChannelId FROM ChannelMembers WHERE UserId = ?)
-		ORDER BY DisplayName LIMIT ? OFFSET ?
-	`
+			ORDER BY DisplayName LIMIT ? OFFSET ?
+		`
+		args = append(args, teamId, limit, offset)
+	} else {
+		query = `
+			SELECT * FROM Channels
+			WHERE (TeamId = ? OR TeamId = '')
+			AND DeleteAt != 0
+			AND Type != ?
+			UNION
+				SELECT * FROM Channels
+				WHERE (TeamId = ? OR TeamId = '')
+				AND DeleteAt != 0
+				AND Type = ?
+				AND Id IN (SELECT ChannelId FROM ChannelMembers WHERE UserId = ?)
+			ORDER BY DisplayName LIMIT ? OFFSET ?
```

Comment:
A bit unrelated to the PR change, but since we're here, isn't that more clear?
```suggestion
                        SELECT * FROM Channels
                        WHERE 
                            (TeamId = ? OR TeamId = '')
                            AND DeleteAt != 0
                            AND (Type != ? 
                            OR (Type = ? AND Id IN (SELECT ChannelId FROM ChannelMembers WHERE UserId = ?)))
                        ORDER BY DisplayName LIMIT ? OFFSET ?
```

Reply:
We'd also need to change the args in line 1520.

Reply:
Great suggestion, it allowed me to use query builder !